### ISSUE TITLE
Request for Debian bullseye/stable-based images

### ISSUE THREAD

**Issue by jmtd** [haskell/docker-haskell#73]
Debian 11 "bullseye" came out in August 2021, and at that point *stable* came to mean Bullseye, and *oldstable* to mean Buster.

hub.docker.com/_/haskell says

>  we support the two most versions of Debian (stable and oldstable) as variants

This is currently not correct. Please consider providing bullseye-based images. Thanks!

**Comment by jmtd** [comment]
It seems cabal-install and ghc provide Debian 10 binaries but not yet 11.

**Comment by AlistairB** [comment]
Yep, my understanding is it will come 'soon' based on the work around https://gitlab.haskell.org/ghc/ghc/-/issues/21002 . Not sure about the cabal side of things, but presumably if GHC is providing it cabal will also follow suit.

**Comment by AlistairB** [comment]
9.4.1 support amd64 deb11, however aarch64 support is not yet included. The images need to support both processor architectures, so we will need to wait. [deb11 aarch64 support](https://gitlab.haskell.org/ghc/ghc/-/issues/22005) is scheduled for 9.4.3.

**Comment by AlistairB** [comment]
cabal 3.8 does not support deb11 unfortunately. If someone is motivated they could raise the issue with the cabal folk.

**Comment by AlistairB** [comment]
I created https://github.com/haskell/cabal/issues/8376

**Comment by ad-si** [comment]
EOL of Buster is approaching, so this should probably be prioritized.
Cabal seems to support Debian 11 now?: https://downloads.haskell.org/~cabal/cabal-install-latest/

Any more blockers? 

![end of life date debian](https://github.com/haskell/docker-haskell/assets/36796532/905516e4-e20e-4ff4-8f4b-fd25e5a9997a)

**Comment by AlistairB** [comment]
No longer a maintainer, but I think GHC is still not providing deb11 for aarch64 https://downloads.haskell.org/~ghc/9.8.1/

**Comment by jmtd** [comment]
Pleased to see deb11 builds have appeared at https://downloads.haskell.org/~ghc/9.10.1-alpha1/
However current debian stable is now Debian 12 ("bookworm").

**Comment by develop7** [comment]
I know, right? Is there anything we could do @jmtd? Say, move to deb12 and use deb11 bindists?

**Comment by develop7** [comment]
Upstream tracking issue: https://gitlab.haskell.org/ghc/ghc/-/issues/25077

**Comment by mauke** [comment]
https://gitlab.haskell.org/ghc/ghc/-/issues/25077 seems to have been resolved.

**Comment by develop7** [comment]
all right, this is misleading now
